### ISSUE TITLE
Better emoji detection

### ISSUE THREAD

**Issue by fxha** [marktext/marktext#1677]
### Description

I think we should restrict emojis (e.g. `:heart:`) to `[a-zA-Z0-9_]` only because it's annoying that text between two `:` is interpreted as emoji. Currently we need to escape the `:` to get the expected behavior.

### Steps to reproduce

```
foo 12:00-14:00 bar
```

**Expected behavior:**

The time should not be interpreted as emoji.

**Actual behavior:**

`:00-14:` is interpreted as wrong emoji.

### Versions

- Mark Text: all
- Operating system: all

**Comment by DanEble** [comment]
My personal preference is that Mark Text do the same as Github—whatever that is, even if it is annoying—because it is also annoying to see a result in one's editor that differs significantly from the result when the markdown is published.

The suggestion needs some revision, as it would reject `:-1:` and `:+1:`.

**Comment by Jocs** [comment]
>My personal preference is that Mark Text do the same as Github

Yes, our gold is to obey the GFM(Github Flavored Markdown), so the emoji  is interpreted also need to be the same as GitHub, so one solution is to validate the text between `:`, only show emoji when it's a valid emoji text.

**Comment by Jocs** [comment]
<img width="984" alt="Screenshot 2019-11-08 at 10 49 49 AM" src="https://user-images.githubusercontent.com/9712830/68682498-2e6ca600-05a0-11ea-8581-85bf3faaa1bd.png">

I marked the text between `:` as orange color, just because I want to show that it's not a validated emoji.but it maybe mislead some user when they input `foo 12:00-14:00 bar`, so how about just remove the orange color?

**Comment by adrienbeau** [comment]
This issue is particularly noticeable because Mark Text underlines standard times, such as 13:30:45. In this example, 30 will be underlined, and this feels wrong.

Many editors require a blank space around the emoji code, to better disambiguate what the user wants.

However Github seems to recognize emoji codes even without blank spaces. So, it might be better to simply remove the orange color.
